packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieve.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { nls } from '../messages';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
+  string
```

Comment:
Right now, the user sees the progress notification on the bottom right hand indicating that the retrieve is running. Do we want to display the notification to the user?